src/composables/useGetRandomId.ts

```diff
@@ -0,0 +1,11 @@
+import { inject } from 'vue'
+import { kongponentsRandomIdFunction } from '@/types'
+import { v4 as uuidv4 } from 'uuid'
+
+export default function useGetRandomId() {
+  const kongponentsId: () => string = inject<() => string>(kongponentsRandomIdFunction, uuidv4)
+
+  return {
+    kongponentsId,
+  }
+}
```

Comment:
suggestion: since there's no reason to export anything else from this file, let's simplify it a bit:
```suggestion
export default inject<() => string>(kongponentsRandomIdFunction, uuidv4)
```

A couple other changes:
1. Let's rename the composable to `useUniqueId`
2. The usage would also change:

    ```ts
    import useUniqueId from '@/composables/useUniqueId' // or maybe just from `@/composables` ?

    const randomTooltipId = useUniqueId()
    ```

Reply:
This doesn't work (I kept getting `useUniqueId is not a function error`)

Reply:
is it getting initialized before the composable is ready?

Reply:
I'm not sure - in the console there were two messages:
```
inject() can only be used inside setup() or functional components
```
and the second one `useUniqueId is not a function`

Reply:
ah wait that's the issue; you can't inject from a composable

Reply:
I tried moving the import inside `<script setup...` but that didn't help

Reply:
You're going to have to inject in every component

Reply:
Does that mean in every component I need to make sure the import statement is within `<script setup...`?

Reply:
Okay let me give it a try

Reply:
Nope, getting the same error. I suggest we live this the way it is - we can always improve this later if we want to

Reply:
I named it `uniqueIdFn` - will that do?

Reply:
sgtm